nltk/stem/snowball.py

```diff
@@ -750,14 +750,19 @@ def stem(self, word):
         self.__checks_1(modified_word)
         # checks2
         self.__checks_2(modified_word)
+        # Pre_Normalization
         modified_word = self.__normalize_pre(modified_word)
+        # Avoid stopwords
+        if modified_word in self.stopwords or len(modified_word) <= 2:
+            return modified_word
+        # Start stemming
         if self.is_verb:
             modified_word = self.__Suffix_Verb_Step1(modified_word)
             if  self.suffixes_verb_step1_success:
                 modified_word = self.__Suffix_Verb_Step2a(modified_word)
                 if not self.suffix_verb_step2a_success :
                     modified_word = self.__Suffix_Verb_Step2c(modified_word)
-                #or next
+                #or next TODO: How to deal with or next instruction
```

Comment:
In which cases would there be more steps that needs to be applied here? Perhaps, it'll be good to list these cases down.

Reply:
We're  working on it and other todos when we solve them we'll send PR for updates 
"or next"  i mean this [line](https://github.com/assem-ch/arabicstemmer/blob/86f9046451e079da40d85106f65cb9194b055aff/algorithm/stemmer.sbl#L586) from the original algorithm as you know i've rewrote the algorithm by hand to follow nltk guideline style code and avoid the generated code from snowball generator.  
You can take a look at this list of [issues and todos](https://github.com/assem-ch/arabicstemmer/issues/1).

Reply:
Could you add a link to the https://github.com/assem-ch/arabicstemmer/issues/1 in the github comment too? That'll be helpful for us to track later. Thanks!

Reply:
I've created an [issue](https://github.com/nltk/nltk/issues/1921) in nltk to track the changes later and added a comment in [https://github.com/assem-ch/arabicstemmer/issues/1](https://github.com/assem-ch/arabicstemmer/issues/1) , I hope this is helpful, sorry for the late replay.